apps/studio/src/server/modules/resource/resource.router.ts

```diff
@@ -663,7 +743,20 @@ export const resourceRouter = router({
   getBatchAncestryWithSelf: protectedProcedure
     .input(getBatchAncestryWithSelfSchema)
     .output(getBatchAncestryWithSelfOutputSchema)
-    .query(async ({ input: { siteId, resourceIds } }) => {
+    .query(async ({ ctx, input: { siteId, resourceIds } }) => {
+      // Validate permissions concurrently for all resourceIds
+      // Note: might want to refactor this if performance becomes an issue
+      await Promise.all(
+        resourceIds.map((resourceId) =>
+          validateUserPermissionsForResource({
+            action: "read",
+            userId: ctx.user.id,
+            siteId: Number(siteId),
+            resourceId: resourceId ? String(resourceId) : null,
+          }),
+        ),
```

Comment:
[question] i checked and this does a single query, which is ok. but i think might be worth just checking in on this w/ around ~1000 resources hahaahha

Reply:
Feels like N+1 query issue, let's avoid DB calls within loops, esp if the loop count can be large

Reply:
update to use bulk validation where only 1 DB query to fetch all resources, but loop the validation on the application-layer (in https://github.com/opengovsg/isomer/pull/1164/commits/506b434a11fdd3e8f1842fd78bf1673c6fd8d9be). Added unit tests here https://github.com/opengovsg/isomer/pull/1164/commits/9d42baad0a7509448393766380defc503dbc11eb